src/sync/app.cpp

```diff
@@ -622,5 +622,74 @@ void App::log_out(std::function<void (Optional<AppError>)> completion_block) con
     log_out(current_user(), completion_block);
 }
 
+void App::switch_user(std::shared_ptr<SyncUser> user,
+                      std::function<void(std::shared_ptr<SyncUser>, Optional<AppError>)> completion_block) const
+{
+    int sync_user_idx = -1;
+    for(size_t i = 0; i < SyncManager::shared().all_users().size(); i++) {
```

Comment:
[style] Simplify this to `auto it = std::find(SyncManager::shared().all_users().begin(), SyncManager::shared().all_users().end(), user)`. Then replace the below checks using the appropriate iterator comparisons, and read the user by dereferencing it from the iterator.

Reply:
done

Reply:
using std::find seems to fail tests on linux, using the traditional for loop structure seems to solve the issues with the tests